### PR TITLE
Configuration via YAML

### LINKED ISSUE #4: Don't expose API Keys on stdout
Hi,
I just saw the following line and it made me concern about security:
https://github.com/narqo/github-weather/blob/86166c02341523e9121820de441f8a57b25bbdfb/main.go#L64

Is it really a good idea to expose these values, a simple message that one of the needed keys wasn't provided or better mention the specific missing key would be IMHO a nice approach to the user.

What was the reason to expose this values?

Cheers!

### PR DESCRIPTION
**Pull request by sleeping-barber**
Hi,
I just played around with the application and introduced to my version a couple of new things:

- made application configurable through YAML file (GitHub, OWM Token, but also Expiration Time and the endpoint of GH due to private GHE)
- added new flag to set the path to configuration file (needed for K8S deployment)
- Adjusted the Manifest files, so user can simply, after applying their desired values, run `kubectl apply -f /deployments`

Fixes #4 😄

Cheers!

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -41,50 +51,60 @@ func main() {
...
+
+	var cfg Config
+	dec := yaml.NewDecoder(f)
+	err = dec.Decode(&cfg)
+	if err != nil {
+		return fmt.Errorf("error parsing configuration file: %v", err)
+	}
+
+	if cfg.Owm.ApiKey == "" || cfg.Github.Token == "" {
+		return fmt.Errorf("no API credentials passed: OpenWeather %q, GitHub %q", cfg.Owm.ApiKey, cfg.Github.Token)
```
Comment (narqo) [inline main.go:81]: I believe in #4 you also wanted to fix this particular line, so the CLI didn't print the secrets, right?

Comment (sleeping-barber) [reply main.go:81]: Yeah, totally forgot about this. Mainly I was talking about the exchange from CLI flags to a config file but I could bake a masked PW feature into this PR.

**Comment by narqo** [comment]
Overall looks awesome. I think I can fix some small things including the comment above myself, a bit later and will release `2.0` tag afterwards. Huge thank you.

**Comment by sleeping-barber** [comment]
Cool. Yeah currently afk mostly :)

**Pull request merged**